pkg/kfilefields/kfilefields.go

```diff
@@ -58,3 +58,19 @@ func ReadFOpForFdType(ft FdType) (uint64, error) {
 	}
 	return ff.FOp, nil
 }
+
+// ReadRealInodeFromFd uses ebpf to read the f_inode pointer from the
+// kernel "struct file" associated with the given fd.
+// Specifically, if fd belongs to overlayFS, it will return the underlying, real inode.
+func ReadRealInodeFromFd(fd int) (uint64, error) {
```

Comment:
Could `ReadRealInodeFromFd` be tested in an unit test? We would not test whether the pointer has the correct value, but rather that the comparison between two realInodePtr is correct. The test would need to setup two temporary overlay filesystems with the same lower layer.

This could be in a separate PR, so this PR is not delayed too much.

Reply:
This should be a unittest for `kfilefields`, if the testing framework could support this, I'd like to do so.